### PR TITLE
no-color

### PR DESCRIPTION
**Pull request by goody-cvs**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Add '--no-color' option to remove color control characters from output for cleaner logs.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ func main() {
 	if err != nil {
 		logger.Error(err.Error())
 	}
+
+        //fmt.Print(reflect.Type(cli.Command))
```
Comment (nimrodkor) [inline main.go:41]: @goody-cvs Please remove unneeded code

File: src/common/reports/report_service.go [hunk 2]
```diff
@@ -18,12 +18,17 @@ type ReportService struct {
...
+	//colorReset  = "\033[0m"
+	//colorGreen  = "\033[32m"
+	//colorYellow = "\033[33m"
+	//colorBlue   = "\033[34m"
+	//colorPurple = "\033[35m"
+	colorReset  = ""
+	colorGreen  = ""
+	colorYellow = ""
+	colorBlue   = ""
+	colorPurple = ""
```
Comment (nimrodkor) [inline src/common/reports/report_service.go:31]: @goody-cvs Can we turn this into a `struct`, and then when creating a new instance we will pass in the bool `noColor` and accordingly set all values? I don't like the global var...

File: src/common/reports/report_service.go [hunk 3]
```diff
@@ -127,38 +149,41 @@ func (r *ReportService) CreateReport() *Report {
...
 	table.SetHeader([]string{"File", "Resource", "Tag Key", "Old Value", "Updated Value", "Yor ID"})
-	table.SetColumnColor(
-		tablewriter.Colors{},
-		tablewriter.Colors{},
-		tablewriter.Colors{tablewriter.Bold},
-		tablewriter.Colors{tablewriter.Normal, tablewriter.FgRedColor},
-		tablewriter.Colors{tablewriter.Normal, tablewriter.FgGreenColor},
-		tablewriter.Colors{},
-	)
+        if ! noColor {
```
Comment (nimrodkor) [inline src/common/reports/report_service.go:177]:
```suggestion
        if !noColor {
```

File: src/common/reports/report_service.go [hunk 4]
```diff
@@ -170,19 +195,21 @@ func (r *ReportService) printUpdatedResourcesToStdout() {
...
 	table.SetRowLine(true)
 	table.SetRowSeparator("-")
-	table.SetColumnColor(
-		tablewriter.Colors{},
-		tablewriter.Colors{},
-		tablewriter.Colors{tablewriter.Bold},
-		tablewriter.Colors{tablewriter.Normal, tablewriter.FgGreenColor},
-		tablewriter.Colors{},
-	)
+        if ! noColor {
```
Comment (nimrodkor) [inline src/common/reports/report_service.go:204]:
```suggestion
        if !noColor {
```

**Review by nimrodkor — COMMENTED** [review]
TY for your contribution @goody-cvs !

File: src/common/reports/report_service.go (hunk 2 above)
Comment (ChanochShayner) [reply src/common/reports/report_service.go:31]: +1, And we won’t need the new func `noColorCheck`

**Review by ChanochShayner — COMMENTED** [review]
@goody-cvs Looks great!
Please fix the broken UT.

**Comment by goody-cvs** [comment]
Remove comments
Change names for slight improvement in clarity
Not sure what UT is, so that func is still there ...

**Comment by ChanochShayner** [comment]
> Remove comments Change names for slight improvement in clarity Not sure what UT is, so that func is still there ...

Looks good.
UT - Unit Tests - [Here](https://github.com/bridgecrewio/yor/actions/runs/3989913737/jobs/6843161366)
You should fix the function's signature in the test files.

File: src/common/reports/report_service.go [hunk 5]
```diff
@@ -127,38 +148,43 @@ func (r *ReportService) CreateReport() *Report {
 // Updated Resources: <int>
 // <New Resources Table> as generated by printNewResourcesToStdout, if not empty
 // <Updated Resources Table> as generated by printUpdatedResourcesToStdout, if not empty
-func (r *ReportService) PrintToStdout() {
-	PrintBanner()
-	fmt.Println(colorReset, "Yor Findings Summary")
-	fmt.Println(colorReset, "Scanned Resources:\t", colorBlue, r.report.Summary.Scanned)
-	fmt.Println(colorReset, "New Resources Traced: \t", colorYellow, r.report.Summary.NewResources)
-	fmt.Println(colorReset, "Updated Resources:\t", colorGreen, r.report.Summary.UpdatedResources)
+func (r *ReportService) PrintToStdout(colorBool bool) {
```
Comment (nimrodkor) [inline src/common/reports/report_service.go:151]: @goody-cvs Why replace the meaningful `noColor` name with a `colorBool` name?
Now when `colorBool` is `false`, it actually means there should be color 😓

Comment (goody-cvs) [reply src/common/reports/report_service.go]: You have a very good point. I reverted that change. Hopefully the new names are more intuitive.

**Review by nimrodkor — APPROVED** [review]

**Review by ChanochShayner — APPROVED** [review]

File: src/common/reports/results_test.go [hunk 6]
```diff
@@ -118,10 +118,11 @@ func TestResultsGeneration(t *testing.T) {
 	t.Run("Test CLI output structure", func(t *testing.T) {
 		ReportServiceInst.CreateReport()
 
-		output := utils.CaptureOutput(ReportServiceInst.PrintToStdout)
+		output := utils.CaptureOutput(ReportServiceInst.PrintToStdout(noColorBool bool))
```
Comment (nimrodkor) [inline src/common/reports/results_test.go:121]: @goody-cvs Should this be like so:
```suggestion
		output := utils.CaptureOutput(ReportServiceInst.PrintToStdout(false))
```
?

File: src/common/reports/report_service.go [hunk 7]
```diff
@@ -127,38 +148,43 @@ func (r *ReportService) CreateReport() *Report {
...
 // <Updated Resources Table> as generated by printUpdatedResourcesToStdout, if not empty
-func (r *ReportService) PrintToStdout() {
-	PrintBanner()
-	fmt.Println(colorReset, "Yor Findings Summary")
-	fmt.Println(colorReset, "Scanned Resources:\t", colorBlue, r.report.Summary.Scanned)
-	fmt.Println(colorReset, "New Resources Traced: \t", colorYellow, r.report.Summary.NewResources)
-	fmt.Println(colorReset, "Updated Resources:\t", colorGreen, r.report.Summary.UpdatedResources)
+func (r *ReportService) PrintToStdout(noColorBool bool) {
+        colors := noColorCheck(noColorBool)
+	PrintBanner(noColorBool)
```
Comment (nimrodkor) [inline src/common/reports/report_service.go:153]: @goody-cvs Can `PrintBanner` get the calculated `colors` or even `colors.Purple`, and avoid computing it too?

**Review by nimrodkor — COMMENTED** [review]
Minor details...

**Review by nimrodkor — APPROVED** [review]

**Comment by nimrodkor** [comment]
Please merge from master, we moved to go 1.19 😓

**Comment by goody-cvs** [comment]
Hi folks, any update on this PR? We'd love to start using this feature without having to use my fork. If I can help, please let me know, thank you.

-Goody

**Comment by ChanochShayner** [comment]
Hey @goody-cvs :)
Thank you for the contribution and the fixes for the requested changes.
Fantastic work!

**Comment by goody-cvs** [comment]
Thank you for the opportunity.

-Goody

From: ChanochShayner ***@***.***>
Reply-To: bridgecrewio/yor ***@***.***>
Date: Thursday, February 16, 2023 at 9:17 AM
To: bridgecrewio/yor ***@***.***>
Cc: "Goodwin, Jason Theodore" ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [bridgecrewio/yor] no-color (PR #332)

**** External Email - Use Caution ****
________________________________
Hey @goody-cvs<https://github.com/goody-cvs> :)
Thank you for the contribution and the fixes for the requested changes.
Fantastic work!

**Pull request merged**